### PR TITLE
validate_regions return all when local

### PR DESCRIPTION
**Pull request by colincadams**

### CODE REVIEW

File: recidiviz/tests/ingest/scrape/ingest_utils_test.py [hunk 1]
```diff
@@ -96,18 +96,13 @@ def test_validate_regions_multiple_all(self, mock_region, _mock_modules):
...
-        mock_get_region.return_value = fake_region
-        fake_region.environment = 'local'
+    def test_validate_regions_multiple_all_invalid(self, _mock_modules):
         assert not ingest_utils.validate_regions(['all', 'invalid'])
 
     @patch('pkgutil.iter_modules',
            return_value=fake_modules('us_ny', 'us_pa', 'us_vt', 'us_pa_greene'))
     def test_validate_regions_empty(self, _mock_modules):
-        assert ingest_utils.validate_regions([]) == []
+        assert ingest_utils.validate_regions([]) = set()
```
Comment (jamwalla) [inline recidiviz/tests/ingest/scrape/ingest_utils_test.py]: `==`

File: recidiviz/ingest/scrape/ingest_utils.py [hunk 2]
```diff
@@ -33,39 +33,47 @@ def lookup_timezone(timezone: Optional[str]) -> Optional[tzinfo]:
...
-def validate_regions(region_list, timezone: tzinfo = None):
+def _regions_matching_environment(region_codes: Set[str]) -> Set[str]:
+    """Filter to regions with the matching environment.
+
+    If we are running locally, include all supported regions.
+    """
+    if not environment.in_gae():
+        return region_codes
+    gae_env = environment.get_gae_environment()
+    return  set(filter(lambda x: regions.get_region(x).environment == gae_env,
```
Comment (jamwalla) [inline recidiviz/ingest/scrape/ingest_utils.py]: is this nicer as a comprehension? up to you!

File: recidiviz/tests/ingest/scrape/ingest_utils_test.py (hunk 1 above)
Comment (colincadams) [reply recidiviz/tests/ingest/scrape/ingest_utils_test.py]: oh wow thanks

File: recidiviz/ingest/scrape/ingest_utils.py (hunk 2 above)
Comment (colincadams) [reply recidiviz/ingest/scrape/ingest_utils.py]: changed to a comprehension for consistency

**Pull request merged**